vars/sectionDeployToAKS.groovy

```diff
@@ -94,8 +101,11 @@ def call(params) {
                 timeoutWithMsg(time: 10, unit: 'MINUTES', action: 'Smoke Test - AKS') {
                   try {
                     builder.smokeTest()
-                  } finally {
                     savePodsLogs(dockerImage, params, "smoke")
+                  } catch(err) {
+                    savePodsLogs(dockerImage, params, "smoke")
+                    clearHelmReleaseForFailure(config, dockerImage, params, pcr)
+                    error 'Build failed. Please check the build logs and the pods logs for further information. You will find the pods logs under Artifacts.'
```

Comment:
Should we log this and throw actual err Thoughts @timja ?

Reply:
I tried with "throw err"... didn't seem to make much of a difference in the logs and I found this to be clearer.

The commit history should show these attempts, I believe.

Reply:
I think this could hide the actual error though especially if something inside is throwing a specific error then that will never be shown I think with this

Reply:
That was my initial thought as well, which is why I tried with "throw err", only to realize the returned error in that case was just a numerical code with no particular value and everything else was identical.

I do however appreciate that this might have been specific to my build, so I'll put "throw err" back in, as from development perspective I also consider it the better option.